src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -714,6 +709,10 @@ public static bool StartsWithPrefix(string prefix, string key)
                     return false;
                 }
             }
+            else if (key.Length == prefix.Length)
+            {
+                return true;
```

Comment:
This should be `return (StringComparer.OrdinalIgnoreCase.Equals(key, prefix))` (or just `string.Equals(...Ordinal.IgnoreCase)`). No?